go.mod

```diff
@@ -10,7 +10,7 @@ require (
 	github.com/bitly/go-simplejson v0.5.0
 	github.com/bmizerany/assert v0.0.0-20160611221934-b7ed37b82869 // indirect
 	github.com/boj/redistore v0.0.0-20160128113310-fc113767cd6b // indirect
-	github.com/btcsuite/btcd v0.0.0-20190115013929-ed77733ec07d
+	github.com/btcsuite/btcd v0.20.1-beta
```

Comment:
what's up with the beta version?

Reply:
go did this by itself

Reply:
I was also suspicious about this so I checked the published versions:
```
    v0.20.1-beta – Nov 13, 2019
    v0.13.0-beta2 – Dec 19, 2018
    v0.13.0-beta – Nov 18, 2018
```
Source: https://pkg.go.dev/github.com/btcsuite/btcd?tab=versions

🤦

Reply:
I want to clarify: this PR adds `libp2p/go-libp2p-core`, which depends on `btcsuite/btcd`:
```
➜  chainlink git:(feature/#174061930-libp2p-keygen) ✗ go mod graph  | grep btcsuite | grep libp2p
github.com/libp2p/go-libp2p-core@v0.6.0 github.com/btcsuite/btcd@v0.20.1-beta
```
`go mod graph` prints the adjacency list of the full dependency graph

As you’ve seen above, the latest version of btcd has a beta tag. 
Also, go will resolve dependencies when a command that compiles the source is executed, that’s `go run`, `go build`, `go install`, `go test`, and maybe others. Sam did a `go test`! 

I thinks there is no error being introduced here, @secure-lorenz  and I were just surprised by the `beta` tag. 

This blog post gives more details about go modules:  https://blog.golang.org/using-go-modules. There is a paragraph in there that explains what is considered the "latest" version:

```
(“Latest” is defined as the latest tagged stable (non-prerelease) version, or else the latest tagged prerelease version, or else the latest untagged version.)
```

Reply:
I want to clarify: this PR adds `libp2p/go-libp2p-core`, which depends on `btcsuite/btcd`:
```
➜  chainlink git:(feature/#174061930-libp2p-keygen) ✗ go mod graph  | grep btcsuite | grep libp2p
github.com/libp2p/go-libp2p-core@v0.6.0 github.com/btcsuite/btcd@v0.20.1-beta
```
`go mod graph` prints the adjacency list of the full dependency graph

As you’ve seen above, the latest version of btcd has a beta tag. 
Also, go will resolve dependencies when a command that compiles the source is executed, that’s `go run`, `go build`, `go install`, `go test`, and maybe others. Sam did a `go test`! 

I thinks there is no error being introduced here and I were just surprised by the `beta` tag. 

This blog post gives more details about go modules:  https://blog.golang.org/using-go-modules. There is a paragraph in there that explains what is considered the "latest" version:

> (“Latest” is defined as the latest tagged stable (non-[prerelease](https://semver.org/#spec-item-9)) version, or else the latest tagged prerelease version, or else the latest untagged version.)